plugins/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -142,6 +185,42 @@
 from ansible_collections.community.general.plugins.module_utils.gitlab import auth_argument_spec, gitlab_authentication
 
 
+def vars_to_variables(vars, module):
+    # transform old vars to new variables structure
+    variables = list()
+    for item in vars.keys():
+        if (isinstance(vars.get(item), string_types) or
+           isinstance(vars.get(item), (integer_types, float))):
+            variables.append(
+                {
+                    "name": item,
+                    "value": str(vars.get(item))
+                }
+            )
+
+        elif isinstance(vars.get(item), dict):
+            new_item = {"name": item, "value": vars.get(item).get('value')}
+
+            if vars.get(item).get('masked'):
+                new_item['masked'] = vars.get(item).get('masked')
+
+            if vars.get(item).get('protected'):
+                new_item['protected'] = vars.get(item).get('protected')
+
+            if vars.get(item).get('environment_scope'):
+                new_item['environment_scope'] = vars.get(item).get('environment_scope')
+
+            if vars.get(item).get('variable_type'):
+                new_item['variable_type'] = vars.get(item).get('variable_type')
+
+            variables.append(new_item)
+
+        else:
+            module.fail_json(msg="value must be of type string, integer, float or dict")
+
+    return variables
```

Comment:
Why not return full dictionaries (with all keys), like the new option does?

Reply:
Do you mean the first `if` block

```py
if (isinstance(vars.get(item), string_types) or
           isinstance(vars.get(item), (integer_types, float))):
```

and why it don't enricht the default properties/attributes?

Good question, I must think about why I did it that way.